### PR TITLE
ModernizeTypesCastingFixer - fix for operators with higher precedence

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4239

### CODE REVIEW

File: src/Fixer/CastNotation/ModernizeTypesCastingFixer.php [hunk 1]
```diff
@@ -138,6 +145,10 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
                     $tokens->removeTrailingWhitespace($functionName);
                 }
 
+                if ($wrapInParenthesises) {
+                    $tokens->insertAt($closeParenthesis, new Token(')'));
```
Comment (guilliamxavier) [inline src/Fixer/CastNotation/ModernizeTypesCastingFixer.php]: Thanks for the reactivity, but what will this result for e.g. `<?php $foo = strval($x + $y)[0];`? What about combining with the existing `$preserveParenthesises`?

Comment (guilliamxavier) [reply src/Fixer/CastNotation/ModernizeTypesCastingFixer.php]: OK I just tried your fix and the result is `<?php $foo = ((string) ($x + $y))[0];`, which is good! :+1: might be worth to add this case to the tests ;)

**Comment by SpacePossum** [comment]
ping @kalessil as original author, if you time for a little review please do so :)

File: src/Fixer/CastNotation/ModernizeTypesCastingFixer.php [hunk 2]
```diff
@@ -110,6 +110,10 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
 
                 $preserveParenthesises = $countParamTokens > 1;
 
+                $afterCloseParenthesisIndex = $tokens->getNextMeaningfulToken($closeParenthesis);
+                $afterCloseParenthesisToken = $tokens[$afterCloseParenthesisIndex];
+                $wrapInParenthesises = $afterCloseParenthesisToken->equals('[') || $afterCloseParenthesisToken->isGivenKind(T_POW);
```
[the remark this answers is not in the document]
Comment (kalessil) [reply src/Fixer/CastNotation/ModernizeTypesCastingFixer.php]: Ah, ok - I see now from the bug-report. If we need to only take care of lines (from the link) before typecasting, then looks good.

And now I wondering if it's time for me to get some sleep or PhpStorm AST tree is incorrect. Let's hope it's the first option ;)

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Pull request merged**